nightfall-client/src/services/transfer.mjs

```diff
@@ -166,7 +167,15 @@ async function transfer(transferParams) {
       secrets.cipherText.flat().map(text => text.field(BN128_GROUP_ORDER)),
       ...secrets.squareRootsElligator2.map(sqroot => sqroot.field(BN128_GROUP_ORDER)),
     ],
-    compressedSecrets.map(text => generalise(text.hex(32, 31)).field(BN128_GROUP_ORDER)),
+    compressedSecrets.map(text => {
+      const bin = text.binary.padStart(256, '0');
+      const parity = bin[0];
```

Comment:
ahh ok gotcha. I remember this about general-number.

So anything with a parity of 0 will be < 256 (since `.binary` removes leading zeroes) and hence needs to be padded - which will correct re-associate the 0 bit.
Otherwise a parity of 1 will have a `.length` == 256 and thus the padding does nothing but we still retrieve the correct bit?